ravenpy/models/importers.py

```diff
@@ -219,78 +219,82 @@ class RoutingProductGridWeightImporter:
 
     CRS_LLDEG = 4326  # EPSG id of lat/lon (deg) coordinate referenence system (CRS)
     CRS_CAEA = 3573  # EPSG id of equal-area    coordinate referenence system (CRS)
-    HRU_ID_FIELD = "HRU_ID"
+    ROUTING_ID_FIELD = "HRU_ID"
+    NETCDF_INPUT_FIELD = "NetCDF_col"
     DIM_NAMES = ("lon", "lat")
     VAR_NAMES = ("lon", "lat")
+    AREA_ERROR_THRESHOLD = 0.05
 
     def __init__(
         self,
-        shapefile_path,
-        nc_data_path,
+        input_file_path,
+        routing_file_path,
         dim_names=DIM_NAMES,
         var_names=VAR_NAMES,
-        hru_id_field=HRU_ID_FIELD,
+        routing_id_field=ROUTING_ID_FIELD,
+        netcdf_input_field=NETCDF_INPUT_FIELD,
         gauge_ids=None,
         sub_ids=None,
+        area_error_threshold=AREA_ERROR_THRESHOLD,
     ):
         self._dim_names = tuple(dim_names)
         self._var_names = tuple(var_names)
-        self._hru_id_field = hru_id_field
+        self._routing_id_field = routing_id_field
+        self._netcdf_input_field = netcdf_input_field
         self._gauge_ids = gauge_ids or []
         self._sub_ids = sub_ids or []
+        self._area_error_threshold = area_error_threshold
 
         assert not (
             self._gauge_ids and self._sub_ids
         ), "Only one of gauge_ids or sub_ids can be specified"
 
-        if Path(shapefile_path).suffix == ".zip":
-            shapefile_path = f"zip://{shapefile_path}"
-        self._shapes = geopandas.read_file(shapefile_path)
-        # Note that we cannot use xarray because it complains about variables and dimensions
-        # having the same name.
-        self._data = nc4.Dataset(nc_data_path)
+        self._input_is_netcdf = True
+
+        input_file_path = Path(input_file_path)
+        if input_file_path.suffix == ".nc":
+            # Note that we cannot use xarray because it complains about variables and dimensions
+            # having the same name.
+            self._input_data = nc4.Dataset(input_file_path)
+        elif input_file_path.suffix == ".zip":
+            self._input_data = geopandas.read_file(f"zip://{input_file_path}")
+            self._input_is_netcdf = False
+        elif input_file_path.suffix == ".shp":
+            self._input_data = geopandas.read_file(input_file_path)
+            self._input_is_netcdf = False
+        else:
+            raise ValueError(
+                "The input file must be a shapefile (.shp or .zip) or NetCDF"
+            )
+
+        routing_file_path = Path(routing_file_path)
+        if routing_file_path.suffix == ".zip":
```

Comment:
Should there be an `elif` for ".shp" and a `ValueError` if it's neither?

Reply:
We could but since the CLI specifies that it must be a shapefile, if it's not we can just let the call to `geopandas.read_file` crash (this is a bit lazy of course).

Reply:
There isn't a special mimetype-like prepending needed for a shape. `geopandas.read_file()` should be able to handle them. Raising a `ValueError` will cause errors that otherwise wouldn't be raised (e.g. user sends a valid GeoJSON, GeoPackage, SQLite, etc.)

Reply:
ok. Was just wondering because we also distinguish between zip and shp for the `input_file_path`

Reply:
@Zeitsperre I'm pretty certaint that `geopandas.read_file` crashes if I do not prepend a `zip://` schema when required though.. you say it does not?

Reply:
No, you are correct, it will crash. I'm referring to why it isn't necessary to perform similar formatting for other files.